core/src/main/java/org/springframework/ldap/filter/ProximityFilter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2005-2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ldap.filter;
+
+/**
+ * A filter for 'proximity'. The following code:
+ *
+ * <pre>
+ * ProximityFilter filter = new ProximityFilter(&quot;cn&quot;, &quot;Some CN&quot;);
+ * System.out.println(filter.encode());
+ * </pre>
+ *
+ * would result in:
+ *
+ * <pre>
+ * (cn~=Some CN)
+ * </pre>
+ *
+ * @author Moritz Kobel
+ */
+public class ProximityFilter extends EqualsFilter {
+
+	private static final String PROXIMITY_SIGN = "~=";
+
+	public ProximityFilter(String attribute, String value) {
+		super(attribute, value);
+	}
+
+	/*
+	 * @see org.springframework.ldap.filter.CompareFilter#getCompareString()
+	 */
+	protected String getCompareString() {
```

Comment:
Instead of implementing this, please call `CompareFilter`'s `protected` constructor, which you will have by rebasing with `main`.